src/NosCore.Core/Networking/WebApiAccess.cs

```diff
@@ -56,29 +78,97 @@ public T Get<T>(string route, ServerConfiguration webApi = null)
             }
 
             var client = new HttpClient();
-            HttpResponseMessage response;
+            var response = new HttpResponseMessage();
             client.BaseAddress = webApi == null ? BaseAddress : new Uri(webApi.ToString());
-            if (Token == null)
+            AssignToken(response, client);
+
+            client.DefaultRequestHeaders.Authorization = new AuthenticationHeaderValue("Bearer", Token);
+            response = client.GetAsync(route).Result;
+            if (response.IsSuccessStatusCode)
             {
-                response = client.PostAsync("api/token/connectserver", Content).Result;
-                if (response.IsSuccessStatusCode)
-                {
-                    Token = response.Content.ReadAsStringAsync().Result;
-                }
-                else
-                {
-                    throw new HttpRequestException(response.Headers.ToString());
-                }
+                return JsonConvert.DeserializeObject<T>(response.Content.ReadAsStringAsync().Result);
             }
 
+            throw new HttpRequestException(response.Headers.ToString());
+        }
+
+        public T Post<T>(string route, object data, ServerConfiguration webApi = null)
+        {
+            if (MockValues.ContainsKey(route))
+            {
+                return (T)MockValues[route];
+            }
+
+            var client = new HttpClient();
+            HttpResponseMessage response = new HttpResponseMessage();
+            client.BaseAddress = webApi == null ? BaseAddress : new Uri(webApi.ToString());
+            AssignToken(response, client);
+
+            var param = new Dictionary<string, string>
+            {
+                { "postedData", JsonConvert.SerializeObject(data) }
+            };
+            var content = new FormUrlEncodedContent(param);
```

Comment:
maybe 
content = new StringContent(content);
content.Headers.ContentType = MediaTypeHeaderValue.Parse("application/json");

Reply:
already tried that, it still doesn't bind the parameter